### PR TITLE
feat: change repo links from github to repo pages

### LINKED ISSUE #3154: Feature: Link repositories to their repo page
### Type of feature

🍕 Feature

### Current behavior

Places where we link a repository to GitHub should navigate to the new Repo Pages

### Suggested solution

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Any components that previously link to repositories on GitHub has been changed to repo pages:

- [x] `<TableRepositoryName>`
- [x] `<CardRepoList>`
- [x] `<RecommendedRepoCard>`

## Related Tickets & Documents

Closes #3154 

## Steps to QA
1. Go through each page on the app
2. Any buttons/link to repositories should go to its repo pages, not GitHub

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: components/molecules/RecommendedRepoCard/recommended-repo-card.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const RecommendedRepoCard = ({ fullName, className }: RecommendedRepoCardProps):
       {data && (
         <>
           <div className="flex justify-between w-full">
-            <a target="_blank" href={`https://github.com/${fullName}`} className="flex items-center gap-1.5">
+            <a target="_blank" href={`/s/${fullName}`} className="flex items-center gap-1.5">
```
Comment (nickytonline) [inline components/molecules/RecommendedRepoCard/recommended-repo-card.tsx:36]: This is a link within the app so no need for `target="_blank"`. As well, it should be using the Next.js `<Link />` component.

```suggestion
            <Link href={`/s/${fullName}`} className="flex items-center gap-1.5">
```

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: components/molecules/RecommendedRepoCard/recommended-repo-card.tsx (hunk 1 above)
Comment (bdougie) [inline components/molecules/RecommendedRepoCard/recommended-repo-card.tsx:36]:
```suggestion
            <Link href={`/s/${fullName}`} className="flex items-center gap-1.5">
```
I think we want link and no need for target="_blank"

File: components/molecules/TableRepositoryName/table-repository-name.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ const TableRepositoryName = ({ avatarURL, fullName }: TableRepositoryNameProps):
   return (
     <div className="flex items-center gap-2.5">
       {/* Avatar */}
-      <a href={`https://www.github.com/${handle}/${name}`} target="_blank">
+      <a href={`/s/${fullName}`} target="_blank">
```
Comment (bdougie) [inline components/molecules/TableRepositoryName/table-repository-name.tsx:17]:
```suggestion
      <Link href={`/s/${fullName}`}>
```

File: components/molecules/TableRepositoryName/table-repository-name.tsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ const TableRepositoryName = ({ avatarURL, fullName }: TableRepositoryNameProps):
       <div className="flex flex-col justify-center">
         <Tooltip content={name}>
           <div className="  text-base text-light-slate-12 tracking-tight max-w-[5.5rem] sm:max-w-[9.3rem] md:max-w-[7rem] whitespace-nowrap overflow-hidden overflow-ellipsis">
-            <a href={`https://www.github.com/${handle}/${name}`} target="_blank">
+            <a href={`/s/${fullName}`} target="_blank">
```
Comment (bdougie) [inline components/molecules/TableRepositoryName/table-repository-name.tsx:32]:
```suggestion
            <Link href={`/s/${fullName}`}">
```

**Review by bdougie — CHANGES_REQUESTED** [review]

**Review by nickytonline — APPROVED** [review]

**Review by bdougie — APPROVED** [review]

**Pull request merged**